src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -318,13 +317,28 @@ uct_rdamcm_cm_ep_set_qp_num(struct rdma_conn_param *conn_param,
         /* create a dummy qp in order to get a unique qp_num to provide to librdmacm */
         status = uct_rdmacm_cm_create_dummy_qp(cep->id, cq, &cep->qp);
         if (status == UCS_OK) {
-            conn_param->qp_num = cep->qp->qp_num;
+            *qpn = cep->qp->qp_num;
         }
     }
 
     return status;
 }
 
+static void uct_rdamcm_cm_ep_destroy_qpn(uct_rdmacm_cm_ep_t *cep)
+{
+    /* Will return without any action if no reserved qpn was created. */
+    uct_rdamcm_cm_ep_destroy_reserved_qpn(cep);
+    /* Will return without any action if no dummy qp was created. */
+    uct_rdmacm_cm_ep_destroy_dummy_qp(cep);
```

Comment:
```
if (cap_supported) {
call_1;
} else {
call_2;
}
```
the same for initialization

Reply:
@evgeny-leksikov Logically we don't need to check capability here because we checked "if (cep->qpn == 0)" in the destroy function. If the cap is not supported, then qpn value is 0, destroy function will do nothing.
To achieve cap here, will need extra code to judge whether the qpn was created(the destoy function can be called with the verbs value is 0 as I mentioned in previous email) and code to get the capability.
I agree that adding the code to achieve and check cap here can make the logic more clear, but I also feel it's not necessary since we can add comments to make the logic more clear.
@yosefe WDYT?

Reply:
@evgeny-leksikov @yosefe Fixed.